design/src/main/java/com/ingresse/design/ui/user/DSUserInfoCard.kt

```diff
@@ -0,0 +1,70 @@
+package com.ingresse.design.ui.user
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.ingresse.design.R
+import com.ingresse.design.helper.ResourcesHelper
+import com.ingresse.design.helper.USER_IMAGE_PREFIX
+import com.ingresse.design.helper.setVisible
+import kotlinx.android.synthetic.main.ds_user_info_card.view.*
+
+class DSUserInfoCard(context: Context, attrs: AttributeSet): ConstraintLayout(context, attrs) {
+    val resHelper = ResourcesHelper(context)
+    var type: DSUserInfoCardType = DSUserInfoCardType.UNDEFINED
+    set(value) {
+        field = value
+        updateButton()
+    }
+
+    init {
+        inflate(context, R.layout.ds_user_info_card, this)
+        val array = context.theme.obtainStyledAttributes(attrs,
+            R.styleable.DSUserInfosCard, 0, 0)
+
+        val typeId = array.getInt(R.styleable.DSUserInfosCard_userCardType,
+            DSUserInfoCardType.UNDEFINED.id)
+
+        type = DSUserInfoCardType.getById(typeId)
+        updateButton()
+
+        array.recycle()
+    }
+
+    /**
+     * Function to set the button action
+     *
+     * @param onClick - Action who will called when click on button
```

Comment:
Outro problema de gramática aqui também. Vale revisar com alguém mais fluente em inglês, mas as vezes abreviar a descrição ajuda, tal como _Callback for button click action_ ou algo do gênero.